services/ingest/tests/test_correlation_engine.py

```
@@ -17,6 +17,29 @@
 )
 
 
+class _FakeNestedTransaction:
+    """Minimal async-context-manager stand-in for SQLAlchemy's
+    AsyncSessionTransaction — begin_nested() itself is a plain sync call
+    that returns this, not a coroutine (see conftest.py's mock_session
+    fixture for the same pattern)."""
+
+    async def __aenter__(self):
```

Ruff (0.16.3)

[warning] Missing return type annotation for special method `__aenter__`

(ANN204)